### PR TITLE
Trying to implement Commit status with "In progress"

### PR DESCRIPTION
**Pull request by fonsecas72**
First let me say that this is my first PR using java. 

I've forked @kmadel work and implemented "In Progress" status using the checkout listener (I'm trying to copy somehow what Github is doing).

Hope you can help me to improve any missing stuff. I really need this functionality (and others to come).

### CODE REVIEW

**Comment by amuniz** [comment]
Getting the same issue than I pointed out in #3, HTTP 400:

```
jul 02, 2016 12:23:48 PM hudson.model.listeners.RunListener report
WARNING: RunListener failed
com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketRequestException: HTTP request error. Status: 400: BAD REQUEST.
{"error": {"fields": {"url": ["Enter a valid URL."]}, "message": "Bad request"}}
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.postRequest(BitbucketCloudApiClient.java:486)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.postRequest(BitbucketCloudApiClient.java:513)
	at com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketCloudApiClient.postBuildStatus(BitbucketCloudApiClient.java:303)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketChangesetCommentNotifier.buildStatus(BitbucketChangesetCommentNotifier.java:49)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketBuildStatusNotifications.createBuildCommitStatus(BitbucketBuildStatusNotifications.java:85)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketBuildStatusNotifications.createPullRequestCommitStatus(BitbucketBuildStatusNotifications.java:112)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketBuildStatusNotifications.sendNotifications(BitbucketBuildStatusNotifications.java:175)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketBuildStatusNotifications.access$000(BitbucketBuildStatusNotifications.java:60)
	at com.cloudbees.jenkins.plugins.bitbucket.BitbucketBuildStatusNotifications$JobCompletedListener.onCompleted(BitbucketBuildStatusNotifications.java:203)
	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:199)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.finish(WorkflowRun.java:503)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:212)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
```

Did you try to manually configure a Multibranch Pipeline with Bitbucket branch source?

**Comment by amuniz** [comment]
Ok, it seems I forgot to setup the Jenkins root URL properly. Re-testing now.

**Comment by amuniz** [comment]
I've fixed an issue with webhooks on pull requests builds (https://github.com/jenkinsci/bitbucket-branch-source-plugin/commit/a8eb0a764a6fcdd90e989730680f111b88936c95).

Can you update your PR with this changes please? (basically merging upstream master into your branch).

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java [hunk 1]
```diff
@@ -150,34 +159,48 @@ private static BitbucketSCMSource lookUpBitbucketSCMSource(final SCMSourceOwner
         }
         return null;
     }
+    
+    private static void sendNotifications(Run<?, ?> build, TaskListener listener)
+    {
```
Comment (amuniz) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java]: Please, follow the code style in already existent code base (no newline for braces).

**Comment by fonsecas72** [comment]
@amuniz Thanks for taking a first look. I've merged upstream master and corrected the code-style.

**Comment by fonsecas72** [comment]
@amuniz Hi, I just wanted to reinforce my willingness to change anything so we can move forward with this :)

**Comment by cecchisandrone** [comment]
When will it be merged and released? We are really waiting for it!

**Comment by amuniz** [comment]
👍 let's merge

**Pull request merged**